docs/vector_search.rst

```diff
@@ -0,0 +1,278 @@
+
+Nrtsearch is based on the popular open-source search library, Lucene. Vector search support was added to Lucene in version 9+. It uses the Hierarchical Navigable Small Worlds (HNSW) graph to traverse vector data to find the most relevant results. It supports vector-only search as well as hybrid (vector + text search).
+
+More information about Lucene's support of vector search can be found at:
+
+- `Lucene Vector Search <https://www.apachecon.com/acna2022/slides/04_lucene_vector_search_sokolov.pdf>`_
+
+In the following sections, we will go through different steps to launch and configure an Nrtsearch cluster with vector search support.
+
+What was Already Supported?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Nrtsearch v0 already supports regular vector fields. However, it's not using any kind of graph data structure under the hood to do advanced KNN lookups in the recall/boosting phase. This data can only be used in the scoring phase to find similarity scores using custom functions/painless scripts.
+
+Launching Cluster with Vector Search Support
+============================================
+
+Estimate Cluster Size
+---------------------
+
+Before launching a cluster, you need to figure out how big the data will be.
+
+The general guidelines to calculate the size introduced by embeddings is to use the following formula::
+
+    Size in bytes = Total Num docs * Float Size * (Embeddings dimensions + hnsw additional storage)
+
+For example, if your index will have around 390M docs with embeddings of 512 dimensions, the total size for this cluster will be around 817GB::
+
+    => ~390M * 4 * (512 + 12) = ~817GB
+
+Note that the above formula is only for vector fields. Other fields in the index will require additional space.
+
+Configuring Cluster
+-------------------
+
+To create a cluster that supports vector search, you can follow the guide to launch regular Nrtsearch clusters with a few minor changes that are discussed below.
+
+Add the following vector search-specific settings to the index configuration under the datastore schema repo::
+
+    # As a top-level property
+    complex-types:
+      <VECTOR FIELD NAME>: vector
+    ...
+
+    # Under field property
+    - name: <VECTOR FIELD NAME>
+      type: VECTOR
+      search: true
+      vectorSimilarity: cosine  # You need to reindex data if this value is changed.
+      vectorIndexingOptions:
+        - type: hnsw
+          hnsw_m: 16  # The number of neighbors each node will be connected to in the HNSW graph, default: 16
+          hnsw_ef_construction: 100  # The number of candidates to track while assembling the list of nearest neighbors for each new node, default: 100
+      vectorDimensions: 512
+
+* Lower `hnsw_m` and `hnsw_ef_construction` values can help with lowering latencies and improving indexing throughput. However, they affect search accuracy.
+* Lower values for `vectorDimensions` help with less disk space and indexing throughput. However, it affects search accuracy.
```

Comment:
Mentioning the max dimension limit would be good

Reply:
Added